04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,69 @@
+const { query } = require('express')
+const Product = require('../models/product')
+
+
+
+const getAllProductsStatic = async (req,res) => {
+    const products = await Product.find({price:{$gt:30}})
+    .sort('price')
+    .select('name price')
+    res.status(200).json({products, nbHits: products.length})
+}
+
+const getAllProducts = async (req,res) => {
+    const {featured,company,name,sort,fields, numericFilters} = req.query;
+    const queryObject = {}
+
+    if(featured){
+        queryObject.featured = featured === 'true'?true:false
+    }
+    if(company) {
+        queryObject.company = company;
+    }
+    if(name){
+        queryObject.name =  {$regex:name, $options: 'i'}
+    }
+    if(numericFilters){
+        const operatorMap = {
+            '>':'$gt',
+            '>=':'$gte',
+            '=':'$eq',
+            '<':'$lt',
+            '<=':'$lte',
+
+        }
+        const regEx = /\b(<|>|>=|=|<=)\b/g
+        let filters = numericFilters.replace(regEx,(match)=>`-${operatorMap[match]}-`)
+    const options = ['price','rating'];
+    filters = filters.split(',').forEach((item)=> {
+        const [field,operator,value] = item.split('-')
+        if(options.includes(field)){
+            queryObject[field] = {[operator]: Number(value)}
+        }
+    })
+    
+        console.log(filters)
+    }
+    console.log(queryObject)
+    let result=Product.find(queryObject);
+    if(sort){
+        const sortList = sort.split(',').join(' ');
+        result = result.sort(sortList)
+    }else{
+        result = result.sort('createdAt')
+    }
+    if(fields){
+        const fieldsList = fields.split(',').join(' ')
+        result = result.select(fieldsList)
+    }
+    const page = Number(req.query.page) || 1
```

Comment:
Nitpick:
Be careful about using `Number` because it can convert a boolean to a value:
`Number(true)` becomes `1`

I'd recommend using [parseInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt) or [parseFloat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat) instead, which parses it correctly:
`parseInt(true, 10)` becomes `NaN` (Not a Number)